python/lsst/ts/observatory/control/auxtel/atcs.py

```diff
@@ -0,0 +1,1753 @@
+
+        Method returns once telescope and dome are in sync.
+        """
+        raise NotImplementedError("Start tracking not implemented yet.")
+
+    async def slew_object(
+        self, name, rot_sky=None, pa_ang=None, rot_pa=0.0, slew_timeout=240.0
+    ):
+        """Slew to an object name.
+
+        Use simbad to resolve the name and get coordinates.
+
+        Parameters
+        ----------
+        name : `str`
+            Target name.
+        rot_sky : `float` or `str`
+            Target sky position angle (deg). Default is `None`, which means
+            use `rot_pa`.
+        pa_ang :  `float` or `str`
+            Set rotator so that the reference position is the parallactic
+            angle. Meaning, if pa_ang=0. the rot_sky is set so that y-axis
+            is aligned with the parallactic angle.
+        rot_pa : `float` or `str`
+            Target rotator position angle (deg). Ignored if `rot_sky` is
+            given (Default = 0).
+        slew_timeout : `float`
+            Timeout for the slew command (second).
+
+        """
+
+        object_table = Simbad.query_object(name)
+
+        if object_table is None:
+            raise RuntimeError(f"Could not find {name} in Simbad database.")
+        elif len(object_table) > 1:
+            self.log.warning(f"Found more than one entry for {name}. Using first one.")
+
+        self.log.info(
+            f"Slewing to {name}: {object_table['RA'][0]} {object_table['DEC'][0]}"
+        )
+
+        # radec_icrs = ICRS(Angle(object_table['RA'][0], unit=u.hour),
+        #                   Angle(object_table['DEC'][0], unit=u.deg))
+
+        await self.slew_icrs(
+            ra=object_table["RA"][0],
+            dec=object_table["DEC"][0],
+            rot_sky=rot_sky,
+            pa_ang=pa_ang,
+            rot_pa=rot_pa,
+            target_name=name,
+            slew_timeout=slew_timeout,
+        )
+
+    async def slew_icrs(
+        self,
+        ra,
+        dec,
+        rot_sky=None,
+        pa_ang=None,
+        rot_pa=0.0,
+        target_name="slew_icrs",
+        slew_timeout=240.0,
+        stop_before_slew=True,
+        wait_settle=True,
+    ):
+        """Slew the telescope and start tracking an Ra/Dec target in ICRS
+        coordinate frame.
+
+        Parameters
+        ----------
+        ra : `float` or `str`
+            Target RA (hour).
+        dec : `float` or `str`
+            Target Dec (deg).
+        rot_sky : `float` or `str`
+            Target sky position angle (deg). Default is `None`, which means
+            use `rot_pa`.
+        pa_ang :  `float` or `str`
+            Set rotator so that the reference position is the parallactic
+            angle. Meaning, if pa_ang=0. the rot_sky is set so that y-axis
+            is aligned with the parallactic angle.
+        rot_pa : `float` or `str`
+            Target rotator position angle (deg). Ignored if `rot_sky` is
+            given (Default = 0).
```

Comment:
So, there are indeed 3 different types of strategies. 

rot_sky : This is the "standard" sky position, the angle between y axis and north. 

pa_ang: This strategy is required for taking optimum spectra with LATISS. If set to zero, the rotator is positioned so that the y axis is aligned with the parallactic angle. 

rot_pa: This strategy is to allow users to select arbitrary rotator angle. This will be the actual rotator position. 

Most of the time users will be interested in using either rot_sky or pa_ang, but there are some cases where the user may want to make sure the rotator is set to a specific position.

Reply:
I don't understand that last one. Is that the rotator mount angle? If so what does "position angle" (I assume that's what "pa" means) have to do with it? Usually that would be called the "moount" or "physical" angle and notated rot_mount/rot_phys/rot_ang. It is potentially useful, though it smears images so it's usually only used when doing in-dome pointing, tracking in el/az.

Please consider an enum for the rotation type and a float which defaults to 0 for the rotation angle. I think it would be easier to understand what is going on and it would eliminate any guesswork about which angle is more important than any other angle.

Another solution is to require that two of the angles be None and the 3rd be a float, but that eliminates the ability to provide a useful default, so I the enum better.

In any case this needs more documentation.

Reply:
yes... `rot_pa` was not a good choice... I guess `rot_pos` would suit it better. I like the idea of the enumeration but that doesn't seem very user-friendly....

Note that it kind of have the behavior you suggested in the end, with the exception that, in case the user does not provide any value, it will set the rotator to zero. Maybe the best behavior would be to set `rot_sky` to zero but that is not an angle that is aways feasible, given the rotator operation range.

Reply:
It makes code slightly harder to write and much easier to read, which is usually considered a worthwhile tradeoff.

I think the only safe way to allow 3 totally different angles (safe in the sense that any reader can understand what is going on) is to require two of them to be None, which makes it impossible to provide a sensible default.

Reply:
I think setting the default behavior to `rot_sky=0` is probably the best approach and to implement the `rottype` enumeration. So by default `rotang=0` and `rottype=0`, meaning by default, slew with sky position zero.

Reply:
I'm not convinced. I think it will make code hard to read, for the reasons stated.

It is much more important to make code easy to read than it is to make code easy to write (though both are worthy goals, of course).

Reply:
Not sure I understood.. you think using the enumeration is not the best option? I thought you preferred the enumeration. My proposal is to use the enumeration but use `rotang=0` and `rottype=0` as default values (where `rottype=0` means `rotang` has the same behavior of `rot_sky`).

Reply:
The reason I want to provide defaults for the rotator is to make the method more user friendly. In general, slewing to RA/Dec assumes `rot_sky=0` the reason I was using `rot_pos=0` as the default has to do with us trying to minimize moving the rotator in the early days.

Reply:
Ease of reading and understanding code is much more important than ease of writing code. (At least for code that needs to be maintained for the long term.) This is true *despite* the fact that you will get more complaints from writers than readers. The number of complaints isn't always a good measure of code quality. Writers tend to be lazy and want to take shortcuts that harm readability and long-term maintainability.

(In fact I would also use that as an argument for making angles be instances of astropy angles. It is not always good to cave to "make it as easy to write as possible").

If you feel you must have a default then here are a few suggestions that I think will not harm readability too much:
- If you use an angle and a rotation type, then only use the default if both are None. If either is specified then both must be specified. That eliminates the issue of "what the heck does "rotangle=5" mean? Unfortunately it means the default is buried in the code, so you have to document it.
- If you use 3 angles, then only use the default if all three are None. If you couple this with "at most one angle can be a float, the rest must be None" then that eliminates the confusing precedence rules. Again...the default is buried in the code and so must be documented.
- Use 3 angles and set sky_ang=0 as the default. The rules are "if either of the other two angles is specified then ignore sky_ang" and "if both other angles are specified then raise an exception". This still allows a clumsy coder to specify sky_ang and one of the other two angles, but that does seem unlikely. Maybe that's what you already had, but if so, I had trouble understanding it from your doc strings.

Reply:
Ok... sounds good... I implemented option 1, 

> If you use an angle and a rotation type, then only use the default if both are None. If either is specified then both must be specified. 

Note that I'll keep the old parameters there for backward compatibility. They will issue a deprecation warning if used.